gulpfile.js

```diff
@@ -11,49 +11,119 @@ var source = require('vinyl-source-stream');
 var pkg;
-// update pkg:
+
+////////////////////////////////////////////////////////////////////////////////
+//------------------------- Helper functions ---------------------------------//
+//----------------------------------------------------------------------------//
+
 function readPackage () {
   pkg = JSON.parse(fs.readFileSync('package.json'));
-  if (pkg.dependencies) {
-    b.external(Object.keys(pkg.dependencies));
-  }
 }
 readPackage();
 
-gulp.task('styles', function () {
-  return gulp.src('app/styles/main.scss')
-    .pipe($.sourcemaps.init())
-    .pipe($.sass({
-      outputStyle: 'nested', // libsass doesn't support expanded yet
-      precision: 10,
-      includePaths: ['.'],
-      onError: console.error.bind(console, 'Sass error:')
-    }))
-    .pipe($.postcss([
-      require('autoprefixer-core')({browsers: ['last 1 version']})
-    ]))
-    .pipe($.sourcemaps.write())
-    .pipe(gulp.dest('.tmp/styles'))
-    .pipe(reload({stream: true}));
+////////////////////////////////////////////////////////////////////////////////
+//------------------------- Callable tasks -----------------------------------//
+//----------------------------------------------------------------------------//
+
+gulp.task('default', ['clean'], function () {
+  gulp.start('build');
+});
+
+gulp.task('serve', ['vendorScripts', 'javascript', 'styles', 'fonts'], function () {
+  browserSync({
+    port: 3000,
+    server: {
+      baseDir: ['.tmp', 'app'],
+      routes: {
+        '/node_modules': './node_modules'
+      }
+    }
+  });
+
+  // watch for changes
+  gulp.watch([
+    'app/*.html',
+    'app/assets/graphics/**/*',
+    '.tmp/assets/fonts/**/*'
+  ]).on('change', reload);
+
+  gulp.watch('app/assets/styles/**/*.scss', ['styles']);
+  // gulp.watch('app/scripts/**/*.js', ['javascript']);
+  gulp.watch('app/assets/fonts/**/*', ['fonts']);
+  gulp.watch('package.json', ['vendorScripts']);
+});
+
+gulp.task('clean', require('del').bind(null, ['.tmp', 'dist']));
+
+gulp.task('build', ['javascript'], function () {
+  gulp.start(['html', 'images', 'fonts', 'extras'], function () {
+    return gulp.src('dist/**/*')
+      .pipe($.size({title: 'build', gzip: true}))
+      .pipe(exit());
+  });
+});
+
+////////////////////////////////////////////////////////////////////////////////
+//------------------------- Browserify tasks ---------------------------------//
+//------------------- (Not to be called directly) ----------------------------//
+//----------------------------------------------------------------------------//
+
+// Compiles the user's script files to bundle.js.
+// When including the file in the index.html we need to refer to bundle.js not
+// main.js
+gulp.task('javascript', function() {
+  var watcher  = watchify(browserify({
+    entries: ['./app/assets/scripts/main.js'],
+    debug: true,
+    cache: {}, packageCache: {}, fullPaths: true
+  }));
+
+  if (pkg.dependencies) {
+    watcher.external(Object.keys(pkg.dependencies));
```

Comment:
@danielfdsilva Just noticed this -- I think that having this here, instead of where we (re)read the package.json file, means that it watchify's list of external modules won't be updated if someone does an `npm install` while gulp is running.